# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1454982326 18000
#      Mon Feb 08 20:45:26 2016 -0500
# Node ID 77a622392714aefdff6bc3cffe1ab5d0c43ba88a
# Parent  25150962bfd3b3f4d81bb65cf901fed5a0a4bcd9
define OCTAVE_EMPTY_CPP_ARG just prior to only use

* ov-base.h (OCTAVE_EMPTY_CPP_ARG): Move definition here.
* oct-conf-post.in.h: From here.

diff --git a/libinterp/octave-value/ov-base.h b/libinterp/octave-value/ov-base.h
--- a/libinterp/octave-value/ov-base.h
+++ b/libinterp/octave-value/ov-base.h
@@ -134,16 +134,18 @@ DEF_CLASS_TO_BTYP (octave_uint16, btyp_u
 DEF_CLASS_TO_BTYP (octave_uint32, btyp_uint32);
 DEF_CLASS_TO_BTYP (octave_uint64, btyp_uint64);
 DEF_CLASS_TO_BTYP (bool, btyp_bool);
 DEF_CLASS_TO_BTYP (char, btyp_char);
 
 // T_ID is the type id of struct objects, set by register_type().
 // T_NAME is the type name of struct objects.
 
+#define OCTAVE_EMPTY_CPP_ARG /* empty */
+
 #define DECLARE_OV_TYPEID_FUNCTIONS_AND_DATA \
   DECLARE_OV_TYPEID_FUNCTIONS_AND_DATA2 (OCTAVE_EMPTY_CPP_ARG)
 
 #define DECLARE_OV_BASE_TYPEID_FUNCTIONS_AND_DATA \
   DECLARE_OV_TYPEID_FUNCTIONS_AND_DATA2(virtual)
 
 #define DECLARE_OV_TYPEID_FUNCTIONS_AND_DATA2(VIRTUAL) \
   public: \
diff --git a/oct-conf-post.in.h b/oct-conf-post.in.h
--- a/oct-conf-post.in.h
+++ b/oct-conf-post.in.h
@@ -112,18 +112,16 @@ typedef unsigned long ino_t;
 #if (SIZEOF_LONG_DOUBLE >= 10) && defined (HAVE_ROUNDL)
 #  define OCTAVE_INT_USE_LONG_DOUBLE
 #  if (SIZEOF_LONG_DOUBLE < 16 \
        && (defined __i386__ || defined __x86_64__) && defined __GNUC__)
 #    define OCTAVE_ENSURE_LONG_DOUBLE_OPERATIONS_ARE_NOT_TRUNCATED 1
 #  endif
 #endif
 
-#define OCTAVE_EMPTY_CPP_ARG
-
 /* oct-dlldefs.h */
 
 /* FIXME: GCC supports visibility attributes as well, even using the
    same __declspec declaration if desired.  The build system should be
    extended to support GCC and visibility attributes.  */
 #if defined (_MSC_VER)
 #  define OCTAVE_EXPORT __declspec(dllexport)
 #  define OCTAVE_IMPORT __declspec(dllimport)
